code/datums/components/deployable_item.dm

```diff
@@ -70,6 +70,12 @@
 		if(!ishuman(user) || HAS_TRAIT(item_to_deploy, TRAIT_NODROP))
 			return
 
+		var/area/area = get_area(location)
+		var/turf/open/placement_loc = location
+		if(istype(item_to_deploy, /obj/item/weapon/shield/riot/marine/deployable) && (!placement_loc.allow_construction || area.area_flags & NO_CONSTRUCTION)) // long ass check to prevent things like deployable shields on alamo
```

Comment:
This is an omega stinky snowflake check, don't do this. Probs best to just make checking the construction flag definable by an arg when setting the component.

Reply:
Alright, I believe I did what ya wanted by using args instead of the snowflake